### ISSUE TITLE
Shallow git fetch/clone

### ISSUE THREAD

**Issue by arafalov** [dweiss/lucene-git-guides#1]
With the latest git, it is possible to commit from a shallow clone. So, it may make sense for new developers to do a shallow (single-branch) clone to start the work and avoid the heavy weight of Lucene/Solr history.

Also, it would be nice to have a suggested workflow for backporting the fixes from _master_ to older release branches.

**Comment by dweiss** [comment]
I didn't know about shallow commit possibility -- interesting! Pulling full history from github is still pretty fast though, isn't it? 

In any case, I don't mind merging if you wish to extend those "guides". Note they're not at all official and no consensus exists on workflows/ processes within Lucene (freedom of choice won over standardization).

**Comment by arafalov** [comment]
I think it is both time and space that is an issue. Especially if working on multiple branches for different versions or patches. Though, in the **latest** git, apparently there is some sort of shared/split tree management.

I'll be happy to contribute back to the guide once I've run through the path several times and am comfortable it works. In a meanwhile, I found them very helpful already as I did not have an SVN workflow before.

**Comment by dweiss** [comment]
> Especially if working on multiple branches for different versions or patches. 

I assume you wish to have different checkouts (in parallel) for those patches? If you only work on one thing at a time then you can store all those branches (and even multiple origins) in one git tree -- they have the same hashes, they'll consume the space once.

Anyway, that's why people like (or hate) git: it permits various ways of working with code.

**Comment by arafalov** [comment]
I am a new committer working - for now - primarily on small bug fixes and regressions. So, I do have multiple trees with different in-progress compiled Solr versions at the same time. This may or may not be a pattern for anybody else. 

The main reason though I mentioned shallow trees is because it helps **new** committers to get started faster with that first checkout/edit/patch cycle. They are unlikely to care about anything but the tip of the trunk branch. That's my reasoning.

**Comment by dweiss** [comment]
bq .The main reason though I mentioned shallow trees is because it helps new committers to get started faster with that first checkout/edit/patch cycle.

I used to work with floppies and dial-up modems, so I have more patience, but I understand your motives :).

**Comment by arafalov** [comment]
I barely missed out on the punch cards (Soviet Russia!) and the modems were a luxury when I started. But I also did 3 years of tech support and learned about average people's patience :-)
